docs/content/v0.20/release.md

```diff
@@ -0,0 +1,60 @@
+
+We will introduce another category of tests, _release tests_. These test a given release, using the published artifacts of that release. These tests will be executables that a contributor invokes, passing the identifier of the desired release on the command line. We will create a GitHub workflow that can be explicitly triggered and will apply the release tests (in the branch that the workflow is directed at) to the latest release in that branch. We will also create a GitHub workflow that is simply told which release to test.
+
+We will maintain a document that lists releases that pass our quality bar. The latest of those is thus the latest stable release. This document is updated in `main` as quality evaluations come in.
+
+We will maintain an examples document that shows users how to exercise a few scenarios using the latest stable release.
+
+## Policy
+
+We aim for all regular releases to be working. In order to do that, we have to make test releases and test them. The widely recognized pattern for doing that is to make "release candidates" (i.e., releases for testing purposes) `1.2.3-rc0`, `1.2.3-rc1`, `1.2.3-rc2`, and so on, while trying to get to a quality release `1.2.3`. Once one of them is judged to be of passing quality, we declare it to be `1.2.3` (i.e., add another tag, `v1.2.3` in addition to the existing tag like `v1.2.3-rc3`).
+
+When reaching a new major or minor version, we make a release branch for it. That is, a branch named `release-$major.$minor`. At first this branch points to the same commit as `main`, and we make release `$major.$minor.0-rc0` from it. After that work can continue, if necessary, on this release branch while work on `main` proceeds to develop the next minor (or major) version.
+
+## Open questions
+
+What to do about a deliberately feature-incomplete release (development snapshot)? Would we be making releases identified `0.20.0-milestone.1` (with `0.20.0-milestone.1.rc.0`, `0.20.0-milestone.1.rc.1` and so on for testing) and `0.20.0-milestone.2` (also with its own test releases) before getting to full-featured rlease `0.20.0`? Or would we simply decleare releases `0.20.0` and `0.21.0` to be deliberately feature-incomplete and release `0.22.0` would be the full-featured release?
+
+What about the clusteradm container image?
+
+Do we publish any PostCreateHook(s)? How does a user use it(them)? I can think of two options for the KCM-for-WDS PCH, neither of which is great. One is simpler but a little more tedious for users: delete the PostCreateHook that installs the KCM for a WDS, instead ask the user to install a Helm chart per WDS. The other option piles on more technology: define a Helm chart that the deployer installs once in the host cluster to define the KCM-for-WDS PCH (and remember: what this thing does is install the KCM-for-WDS Helm chart).
+
```

Comment:
PostCreateHooks for kubestellar use are now part of the KubeStellar git repo, so they would be tagged with the release. Why can't we just require to do the step

```
kubectl --context kind-kubeflex  apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/<release-tag>/config/postcreate-hooks/kubestellar.yaml
kubectl --context kind-kubeflex  apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/<release-tag>/config/postcreate-hooks/ocm.yaml
```

and then the user can just use `kflex create wds<n> -p kubeflex` how many times he/she wants to create the WDS and deploy the controller ?

Reply:
Because a specific version is needed in such commands. But we want no self-references.

Reply:
Yes, I think that would work.

Reply:
@pdettori : So the PostCreateHook source in GitHub in release `1.2.3` has the string "1.2.3" in it? That is a self-reference. It means that each commit has to refer to a release that does not yet exist when the commit is made, and each commit can be in only one release.

Reply:
Understood. One solution is what you outlined with publishing a helm chart to install the two PCHs where we use the same approach to tagging used with the other helm charts to avoid the self referencing.

Reply:
Yes, that is what I was trying to say in the second approach listed.

Reply:
I am in the process of testing a solution for https://github.com/kubestellar/kubeflex/issues/215, which would eliminate the need to create an additional helm chart.